### DISCUSSION TITLE
Licensing and advertising as open source

### DISCUSSION THREAD

**Discussion by ssddanbrown** [swiftwave-org/swiftwave#680] [General]
Hello, 
Just came across your project [on Reddit](https://www.reddit.com/r/selfhosted/comments/1d4c6l4/swiftwave_v2_simple_lightweight_open_source/) where I commented about the license not being widely considered open source so thought I'd drop in here to add extra context. 

The added license terms, as is, goes against the [commonly considered open source definition](https://opensource.org/osd) due to the limitations these terms add on use/modification/distribution. 
It looks like you maybe copied these terms from dokploy? So therefore if you want detail into why these terms are conflicting with the OSD I've written detail on this in [my query to dokploy here](https://github.com/Dokploy/dokploy/discussions/3). _(Note: Their updated licensing is still problematic so I'd avoid copying that too)._

Additionally, as mentioned in that linked dokploy thread, modifying the Apache license like this goes against [Apache's own terms of using their license](https://www.apache.org/foundation/license-faq.html#mod-license) and can cause confusion since you're not really providing the project/code under the Apache 2.0 terms once the additional terms are considered.

To be clear, I respect your rights to license your efforts in whatever way works for you, but I think it's important not to be potentially misleading when it comes to advertising as open source and/or when advertising the license/terms that a project is provided under.

**Comment by tanmoysrt** [comment]
Thanks @ssddanbrown for pointing this out.

I have actually find out in Caprover, which is also a popular solution. They use also use same, https://github.com/caprover/caprover?tab=License-1-ov-file .

So, I thought it may be permissible to use as it doesn't pose more other restrictions like GPLv3. Although the license was added at the initial stage and I haven't looked back in it.

I don't want to go to GPL or AGPL, as that also restrict modification for private use, So I will make it just Apache 2.0.

**Reply by ssddanbrown** [reply]
Ah, okay, I'll look into that caprover scenario and query that too if not already done so.

> I don't want to go to GPL or AGPL, as that also restrict modification for private use, So I will make it just Apache 2.0.

Just a point on that, The GPL/AGPL do not really restrict modification (or use) for private use. These licenses may have extra requirements when it comes to _**distribution**_ (Primarily to ensure the user being distributed to gets the project/source under the same rights/terms), with the AGPL considering network access/use as distribution.

**Reply by tanmoysrt** [reply]
Yup, but people hesitate to use any tool that use GPL. Although most people going to install it and just run it. Very few people may require to modify it for any special usage.

So, I thought, it's better to keep it Apache 2.0. If anytime in future, I want to release some paid feature, I can just provide those in a different way without disclosing the source code of the particular features. So, we can just keep `Swiftwave`'s all feature openly available.

**Comment by tanmoysrt** [comment]
Fixed https://github.com/swiftwave-org/swiftwave/pull/682

Thanks @ssddanbrown again for pointing out the issue

**Reply by ssddanbrown** [reply]
:partying_face: thanks for taking on, and being receptive to the feedback!

**Comment by gabrielcossette** [comment]
Awesome! 🎉

Thanks @ssddanbrown for being so proactive regarding OSS licensing with self-hosted solutions in general, and thank you @tanmoysrt for being so responsive indeed!
